app/Policies/UserPolicy.php

```diff
@@ -47,9 +47,9 @@ public function isAdmin(?User $user): bool
 	 *
 	 * @return bool
 	 */
-	public function canEditSettings(User $user): bool
+	public function mayEditOwnSettings(User $user): bool
 	{
-		return !$user->is_locked;
+		return $user->may_edit_own_settings;
 	}
```

Comment:
I haven't tested it yet, but we should have something like

```php
public function canEdit(User $user, User $userToBeEdited): bool
{
    return $user->id === $userToBeEdited->id && $user->may_edit_own_settings;
}

We do not need the check for `is_administrate` as I wrote above, because this is already handled in `before()`.

Reply:
I think this is over kill: the reason being that only admins can edit other users. This is handled by the `before()` .